sencha-workspace/SlateTasksStudent/app/controller/Dashboard.js

```diff
@@ -136,6 +148,49 @@ Ext.define('SlateTasksStudent.controller.Dashboard', {
         });
     },
 
+    onStudentSelectorSelect: function(combo, rec) {
+        var me = this,
+            studentCombo = me.getStudentSelectorCombo(),
+            params = me.getSearchParams(),
+            username;
+
+        if (!studentCombo) {
+            return;
+        }
+
+        // skip rerouting if student selection hasn't changed
+        username = studentCombo.getValue();
+        if (username === params.student) {
+            return;
+        }
+
+        // reroute page to reflect selected student
+        params.student = username;
+        location.search = Ext.Object.toQueryString(params);
+    },
+
+    onStudentSelectorBoxReady: function(combo) {
```

Comment:
don't use `boxready` for this, what you're doing isn't actually dependent on the combobox having it's width/height calculated, this is a false causal chain and thar lie the path to chaos and jank. Do it up in `onLaunch` instead where you're already grabbing a reference to it. The component tree and API will be available as soon as you `this.getDashboard()` instantiates it on the first line of `onLaunch`